components/concept/sync/src/main/java/mozilla/components/concept/sync/OAuthAccount.kt

```diff
@@ -33,20 +33,155 @@ data class AuthFlowUrl(val state: String, val url: String)
  */
 @SuppressWarnings("TooManyFunctions")
 interface OAuthAccount : AutoCloseable {
+
+    /**
+     * Constructs a URL used to begin the OAuth flow for the requested scopes and keys.
+     *
+     * @param scopes List of OAuth scopes for which the client wants access
+     * @return Deferred AuthFlowUrl that resolves to the flow URL when complete
+     */
     fun beginOAuthFlowAsync(scopes: Set<String>): Deferred<AuthFlowUrl?>
+
+    /**
+     * Constructs a URL used to begin the pairing flow for the requested scopes and pairingUrl.
+     *
+     * @param pairingUrl URL string for pairing
+     * @param scopes List of OAuth scopes for which the client wants access
+     * @return Deferred AuthFlowUrl Optional that resolves to the flow URL when complete
+     */
     fun beginPairingFlowAsync(pairingUrl: String, scopes: Set<String>): Deferred<AuthFlowUrl?>
+
+    /**
+     * Returns current FxA Device ID for an authenticated account.
+     *
+     * @return Current device's FxA ID, if available. `null` otherwise.
+     */
     fun getCurrentDeviceId(): String?
+
+    /**
+     * Returns session token for an authenticated account.
+     *
+     * @return Current account's session token, if available. `null` otherwise.
+     */
     fun getSessionToken(): String?
+
+    /**
+     * Returns an authorized auth code for the passed scopes
```

Comment:
Let's rephrase that a little bit to something like `Provisions a scoped OAuth code for a given [clientId].`